src/main/java/anya/Anya.java

```diff
@@ -0,0 +1,187 @@
+                    throw new AnyaException("Anya doesn't understand this command.");
+                }
+
+            } catch (AnyaException e) {
+                this.ui.errorMessage(e.getMessage());
+            }
+
+            userInput = this.ui.readLine();
+        }
+
+        // Exit
+        try {
+            this.ui.savingFileMessage();
+            this.storage.saveFile(tasks);
+            this.ui.saveFileSuccessMessage();
+        } catch (IOException e) {
+            this.ui.errorMessage(e.getMessage() + " Sorry, Anya is unable to save data.");
+        }
+        this.ui.closeScanner();
+        this.ui.exitMessage();
+    }
+
+    public static void main(String[] args) {
+        new Anya("data/Anya.txt").run();
+    }
+
+    /**
+     * Appends a Task to the end of the TaskList.
+     *
+     * @param tasks A collection of Tasks.
+     * @param task The task to be added into the TaskList.
+     */
+    public void addTask(TaskList tasks, Task task) {
+        tasks.addTask(task);
+        this.ui.addTaskMessage(task, tasks.getLength());
+    }
+
+    /**
+     * Prints out the tasks in the TaskList
+     *
+     * @param tasks A collection of Tasks.
+     */
+    public void list(TaskList tasks) {
+        this.ui.getListMessage(tasks);
+    }
+
+    /**
+     * Marks the task at the specified position as complete.
+     * The TaskList is One-Indexed.
+     *
+     * @param tasks A collection of Tasks.
+     * @param index The index of the task in the TaskList to be marked.
+     */
+    public void mark(TaskList tasks, int index) {
+        Task task = tasks.getTaskFromIndex(index);
+        task.markDone();
+        this.ui.markTaskMessage(task);
+    }
+
+    /**
+     * Marks the task at the specified position as incomplete.
+     * The TaskList is One-Indexed.
+     *
+     * @param tasks A collection of Tasks.
+     * @param index The index of the task in the TaskList to be unmarked
+     */
+    public void unmark(TaskList tasks, int index) {
+        Task task = tasks.getTaskFromIndex(index);
+        task.markUndone();
+        this.ui.unmarkTaskMessage(task);
+    }
+
+    /**
+     * Removes the task at the specified position in the TaskList.
+     * The TaskList is One-Indexed.
+     *
+     * @param tasks A collection of Tasks.
+     * @param index The index of the task in the TaskList to be removed
+     */
+    public void delete(TaskList tasks, int index) {
+        Task removedTask = tasks.getTaskFromIndex(index);
+        tasks.deleteTaskFromIndex(index);
+        this.ui.deleteTaskMessage(removedTask);
+    }
+
+    /**
+     * Gets a filtered TaskList where each task contains the keyword.
+     *
+     * @param tasks A collection of Tasks.
+     * @param keyword The word that the task must contain.
+     */
+    public void find(TaskList tasks, String keyword) {
+        TaskList filteredTasks = tasks.getMatchingTasks(keyword);
+        this.ui.filteredTaskMessage(filteredTasks, keyword);
+    }
+}
```

Comment:
Would it be better to separate the methods handling the commands into a few Command classes?